sdk/nodejs/runtime/resource.ts

```diff
@@ -154,6 +154,50 @@ export function readResource(res: Resource, t: string, name: string, props: Inpu
     }), label);
 }
 
+// The shape of data returned by the built-in invoke to `pulumi:pulumi:readStackResource`.
+interface StackResourceResult {
+    urn: string;
+    // The output properties of the resource represented by `urn` as registered in the Pulumi
+    // engine.
+    outputs: Record<string, any>;
+}
+
+/**
+ * Gets an existing custom resource's state from the engine.  Assumes that the resource has already
+ * been registered by some other code, and looks it up by URN.
+ */
+export function getResource(res: Resource, t: string, name: string, custom: boolean, props: Inputs, opts: ResourceOptions): void {
+    const urn: Input<URN> | undefined = opts.urn;
```

Comment:
Just noting that the deps on this input won't affect the deps of the resource, as it is being read and not registered.

Reply:
That's right. 

I *think* that is the "correct" behaviour.  The actual resource itself was built (and registered) through another path, and the result established here won't be passed back into the engine or serialized anywhere. 

Can you think of any reason why we would need or want dependencies to flow from this `urn` into the resulting resource?

This is of course modeled after `readResource` which behaves in the same way currently.